bob-common/src/mapper.rs

```diff
@@ -130,19 +171,31 @@ impl Virtual {
         let target_nodes = self.get_target_nodes_for_key(key);
         trace!("extract indexes of target nodes");
         trace!("nodes available: {}", self.nodes.len());
-        let offset = self.support_nodes_offset.fetch_add(1, Ordering::Relaxed);
-        let mut support_nodes = Vec::with_capacity(count);
-        for (id, node) in self.nodes.iter().skip(offset % self.nodes.len()) {
-            if target_nodes.iter().all(|i| i.index() != *id) {
+        let mut support_nodes: Vec<&Node> = Vec::with_capacity(count);
+
+        let starting_index = Self::find_support_node_offset(
```

Comment:
Write the comment that the idea of the current logic is to preserve uniform distribution of aliens

Reply:
Done.